### PR TITLE
Fix live Key Vault JCA tests.

### PR DESCRIPTION
**Pull request by vcolin7**
This PR attempts to fix failing live tests such as [these](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1069808&view=logs&j=8eaa2c66-493a-5da3-810a-ade20a7de350&t=70fc4294-596d-517c-9811-9b9f7ffe2e10&l=3163) by exporting Key Vault JCA's implementation package to the test package.

### CODE REVIEW

File: sdk/keyvault/azure-security-keyvault-jca/src/main/java/module-info.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+module azure.security.keyvault.jca {
```
Comment (alzimmermsft) [inline sdk/keyvault/azure-security-keyvault-jca/src/main/java/module-info.java:4]:
```suggestion
module com.azure.security.keyvault.jca {
```

**Review by alzimmermsft — APPROVED** [review]

File: sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/KeyVaultJcaProvider.java [hunk 2]
```diff
@@ -40,7 +40,7 @@
     /**
      * Stores the version.
      */
-    private static final Double VERSION = 1.0;
```
Comment (vcolin7) [inline sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/KeyVaultJcaProvider.java:43]: Made this change so we wouldn't use a deprecated API in line 49.

File: sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/KeyVaultTrustManagerFactoryProvider.java [hunk 3]
```diff
@@ -30,7 +30,7 @@
     /**
      * Stores the version.
      */
-    private static final Double VERSION = 1.0;
```
Comment (vcolin7) [inline sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/KeyVaultTrustManagerFactoryProvider.java:33]: Made this change so we wouldn't use a deprecated API in line 39.

File: sdk/keyvault/azure-security-keyvault-jca/pom.xml [hunk 4]
```diff
@@ -129,14 +129,14 @@
...
-                                    <shadedPattern>com.azure.security.keyvault.jca.implementation.shaded.org.slf4j</shadedPattern>
-                                </relocation>
+                              <relocation>
+                                  <pattern>mozilla</pattern>
+                                  <shadedPattern>com.azure.security.keyvault.jca.implementation.shaded.mozilla</shadedPattern>
+                              </relocation>
+                              <relocation>
+                                  <pattern>org.slf4j</pattern>
+                                  <shadedPattern>com.azure.security.keyvault.jca.implementation.shaded.org.slf4j</shadedPattern>
+                              </relocation>
```
Comment (chenrujun) [inline sdk/keyvault/azure-security-keyvault-jca/pom.xml:139]: Could you please use `4 spaces` as indent in this whole file?

File: sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/KeyVaultJcaProvider.java (hunk 2 above)
Comment (chenrujun) [reply sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/KeyVaultJcaProvider.java:43]: It's deprecated in Java 11. But it's not deprecated in Java 8, we need to support Java 8. So please revert this line. Other wise the pipeline will failed: https://dev.azure.com/azure-sdk/public/_build/results?buildId=1075308&view=logs&j=b70e5e73-bbb6-5567-0939-8415943fadb9&t=b5f103f3-7317-514c-e60c-6e19642365f6&l=17

File: sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/KeyVaultTrustManagerFactoryProvider.java (hunk 3 above)
Comment (chenrujun) [reply sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/KeyVaultTrustManagerFactoryProvider.java:33]: Same here.

File: sdk/keyvault/azure-security-keyvault-jca/src/main/resources/module-info.java [hunk 5]
```diff
@@ -20,4 +20,5 @@
     requires java.logging;
 
     exports com.azure.security.keyvault.jca;
+    exports com.azure.security.keyvault.jca.implementation to com.azure.security.keyvault.jca.test;
```
Comment (chenrujun) [inline sdk/keyvault/azure-security-keyvault-jca/src/main/resources/module-info.java:23]: @ZhuXiaoBing-cn  will move related tests, so this line will not necessary any more after #23709 merged.

**Review by chenrujun — CHANGES_REQUESTED** [review]

**Pull request closed without merging**